tests/specs/lemmas.k

```diff
@@ -88,118 +94,105 @@ module LEMMAS
-    rule _ *Int 0 => 0
+    rule 1 *Int N => N  [simplification]
+    rule N *Int 1 => N  [simplification]
+    rule 0 *Int _ => 0  [simplification]
+    rule _ *Int 0 => 0  [simplification]
 
-    rule N /Int 1 => N
+    rule N /Int 1 => N  [simplification]
 
-    rule 0 |Int N => N
-    rule N |Int 0 => N
-    rule N |Int N => N
+    rule 0 |Int N => N  [simplification]
+    rule N |Int 0 => N  [simplification]
+    rule N |Int N => N  [simplification]
 
-    rule 0 &Int N => 0
-    rule N &Int 0 => 0
-    rule N &Int N => N
-
-    //orienting symbolic term to be first, converting -Int to +Int for concrete values.
-    rule I +Int B => B          +Int I when #isConcrete(I) andBool notBool #isConcrete(B)
-    rule A -Int I => A +Int (0 -Int I) when notBool #isConcrete(A) andBool #isConcrete(I)
-
-    rule (A +Int I2) +Int I3 => A +Int (I2 +Int I3) when notBool #isConcrete(A) andBool #isConcrete(I2) andBool #isConcrete(I3)
-
-    rule I1 +Int (B +Int I3) => B +Int (I1 +Int I3) when #isConcrete(I1) andBool notBool #isConcrete(B) andBool #isConcrete(I3)
-    rule I1 -Int (B +Int I3) => (I1 -Int I3) -Int B when #isConcrete(I1) andBool notBool #isConcrete(B) andBool #isConcrete(I3)
-    rule (I1 -Int B) +Int I3 => (I1 +Int I3) -Int B when #isConcrete(I1) andBool notBool #isConcrete(B) andBool #isConcrete(I3)
-
-    rule I1 +Int (I2 +Int C) => (I1 +Int I2) +Int C when #isConcrete(I1) andBool #isConcrete(I2) andBool notBool #isConcrete(C)
-    rule I1 +Int (I2 -Int C) => (I1 +Int I2) -Int C when #isConcrete(I1) andBool #isConcrete(I2) andBool notBool #isConcrete(C)
-    rule I1 -Int (I2 +Int C) => (I1 -Int I2) -Int C when #isConcrete(I1) andBool #isConcrete(I2) andBool notBool #isConcrete(C)
-    rule I1 -Int (I2 -Int C) => (I1 -Int I2) +Int C when #isConcrete(I1) andBool #isConcrete(I2) andBool notBool #isConcrete(C)
-
-    rule I1 &Int (I2 &Int C) => (I1 &Int I2) &Int C when #isConcrete(I1) andBool #isConcrete(I2) andBool notBool #isConcrete(C)
+    rule 0 &Int N => 0  [simplification]
+    rule N &Int 0 => 0  [simplification]
+    rule N &Int N => N  [simplification]
 
     // 0xffff...f &Int N = N
     rule MASK &Int N => N  requires MASK ==Int (2 ^Int (log2Int(MASK) +Int 1)) -Int 1 // MASK = 0xffff...f
                             andBool 0 <=Int N andBool N <=Int MASK
+      [simplification]
 
     // N &Int 0xffff...f = N
     rule N &Int MASK => N  requires MASK ==Int (2 ^Int (log2Int(MASK) +Int 1)) -Int 1 // MASK = 0xffff...f
                             andBool 0 <=Int N andBool N <=Int MASK
+      [simplification]
 
     // for gas calculation
-    rule A -Int (#if C #then B1 #else B2 #fi) => #if C #then (A -Int B1) #else (A -Int B2) #fi
+    rule A -Int (#if C #then B1 #else B2 #fi) => #if C #then (A -Int B1) #else (A -Int B2) #fi  [simplification]
 
-    rule (#if C #then B1 #else B2 #fi) -Int A => #if C #then (B1 -Int A) #else (B2 -Int A) #fi
-        when notBool #isConcrete(A) andBool #notKLabel(A, "#if_#then_#else_#fi_K-EQUAL")
```

Comment:
github diff doesn't work smartly here.  so, it seems that this lemma is dropped, and all the changes below are simply adding the attribute, right?  it is fine to drop this lemma for now.

Reply:
Yes, I dropped it. Yes, the diff is not helpful, it's easier to follow the git commits.

I'm also adding 4 new lemmas below, and I'm deleting `#notKLabel`'s definition. The latter causes other tests to fail in the continuous build.